File: app/src/main/java/com/taitsmith/busboy/fragments/BusDetailFragment.java
package com.taitsmith.busboy.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;

import com.taitsmith.busboy.R;

import com.taitsmith.busboy.viewmodels.BusDetailViewModel;

public class BusDetailFragment extends Fragment {

    private BusDetailViewModel mViewModel;

    public static BusDetailFragment newInstance() {
        return new BusDetailFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_bus_detail, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = ViewModelProviders.of(this).get(BusDetailViewModel.class);
        // TODO: Use the ViewModel
    }

}

File: app/src/main/java/com/taitsmith/busboy/ui/BusDetailFragment.kt
package com.taitsmith.busboy.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import com.taitsmith.busboy.data.Bus
import com.taitsmith.busboy.databinding.FragmentBusDetailBinding

class BusDetailFragment : DialogFragment() {
    private var _binding: FragmentBusDetailBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentBusDetailBinding.inflate(inflater, container, false)
        binding.bus = requireArguments().getSerializable(ARG_BUS) as Bus
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        private const val ARG_BUS = "bus"

        //pass the Bus through the arguments Bundle (Bus is Serializable) rather than a constructor
        //parameter, so the dialog survives recreation (rotation / process death) instead of the
        //FragmentManager failing to re-instantiate a fragment that has no no-arg constructor.
        fun newInstance(bus: Bus): BusDetailFragment = BusDetailFragment().apply {
            arguments = Bundle().apply { putSerializable(ARG_BUS, bus) }
        }
    }
}
